nsqlookup/resolver.go

```diff
@@ -134,9 +133,78 @@ func (r *ConsulResolver) Resolve(ctx context.Context) (list []string, err error)
 		address = "http://" + address
 	}
 
-	if req, err = http.NewRequest("GET", address+"/v1/catalog/service/"+service, nil); err != nil {
+	// get list of check results for service
+	var checksResults []struct {
+		Node string
+	}
+
+	err = r.getConsul(ctx, fmt.Sprintf("v1/health/checks/%s?passing", service), &checksResults)
+	if err != nil {
 		return
 	}
+
+	// get list of nodes for service
+	var serviceResults []struct {
+		Node           string
+		Address        string
+		ServiceAddress string
+		ServicePort    int
+	}
+
+	err = r.getConsul(ctx, fmt.Sprintf("v1/catalog/service/%s", service), &serviceResults)
+	if err != nil {
+		return
+	}
+
+	list = make([]string, 0, len(checksResults))
+
+	for _, r := range serviceResults {
+		var passing bool
+		for _, c := range checksResults {
+			if c.Node == r.Node {
+				passing = true
+				break
+			}
+		}
+
+		if passing {
+			host := r.ServiceAddress
+			port := r.ServicePort
+
+			if len(host) == 0 {
+				host = r.Address
+			}
+
+			list = append(list, net.JoinHostPort(host, strconv.Itoa(port)))
+		}
+	}
+
+	return
+}
+
+func (r *ConsulResolver) getConsul(ctx context.Context, endpoint string, result interface{}) error {
```

Comment:
This method name's not intuitively telling much about what's going on, how about just naming it `get` since it's making a GET request?

"Consul" in the name is redundant since it's a method of the ConsulResolver type.